src/main/kotlin/fi/metatavu/metaform/server/rest/translate/UserTranslator.kt

```diff
@@ -0,0 +1,61 @@
+package fi.metatavu.metaform.server.rest.translate
+
+import fi.metatavu.metaform.api.spec.model.User
+import fi.metatavu.metaform.api.spec.model.UserFederatedIdentity
+import fi.metatavu.metaform.api.spec.model.UserFederationSource
+import fi.metatavu.metaform.keycloak.client.models.UserRepresentation
+import java.util.UUID
+import javax.enterprise.context.ApplicationScoped
+
+/**
+ * Translator for Users
+ */
+@ApplicationScoped
+class UserTranslator {
+
+    /**
+     * Translates Metaform KC user representation into REST User
+     *
+     * @param entity User Representation
+     * @return REST User
+     */
+    fun translate(entity: UserRepresentation): User {
+        val isUserFederated = !entity.federatedIdentities.isNullOrEmpty()
+        val federatedIdentities =
+            if (isUserFederated) {
+                listOf(UserFederatedIdentity(
+                    source = UserFederationSource.CARD,
+                    userId = entity.federatedIdentities?.get(0)?.userId!!,
+                    userName = entity.federatedIdentities[0].userName!!
+                ))
```

Comment:
Use proper mapping for the source type instead of assuming first is always card.

Reply:
Done